Views/MenuWidget.cshtml

```diff
@@ -0,0 +1,83 @@
+    const string dropdownItemElementName = blockName + "__dropdownItem";
+
+    var contentIdName = FormattableString.Invariant($"{contentElementName}_{Guid.NewGuid()}");
+    var dropdownIdName = FormattableString.Invariant($"{dropdownElementName}_{Guid.NewGuid()}");
+}
+
+@if (!viewModel.NoWrapper)
+{
+    @:<nav class="navbar navbar-expand-md">
+}
+    <div class="container-fluid">
+        <button class="navbar-toggler @togglerElementName"
+                type="button"
+                data-bs-toggle="collapse"
+                data-bs-target="#@contentIdName"
+                aria-controls="@contentIdName"
+                aria-expanded="false"
+                aria-label="@T["Toggle Navigation"]">
+            <span class="navbar-toggler-icon fas fa-bars"></span>
+        </button>
+        <div id="@contentIdName" class="collapse navbar-collapse @contentElementName">
+            <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+                @foreach (var menuItem in viewModel.MenuItems)
+                {
+                    @if (menuItem.Items.Any())
+                    {
+                        <li class="nav-item dropdown">
+                            <a id="@dropdownIdName"
+                               class="nav-link dropdown-toggle @dropdownElementName"
+                               href="#"
+                               role="button"
+                               data-bs-toggle="dropdown"
+                               aria-expanded="false">@menuItem.Text</a>
+                        </li>
+                        <ul class="dropdown-menu" aria-labelledby="@dropdownIdName">
+                            @foreach (var submenu in menuItem.Items)
+                            {
+                                <li>
+                                    <a id="@submenu.Id"
+                                       class="dropdown-item @dropdownItemElementName"
+                                       @if (!submenu.LocalNav)
+                                       {
+                                           <text>target="_blank"</text>
+                                       }
+                                       href="@submenu.Href">@submenu.Text</a>
+                                </li>
+                            }
+                        </ul>
+                    }
+                    else
+                    {
+                        var activeClass = Context.Request.Path == menuItem.Href?.Partition("?").Left ? "active" : string.Empty;
+                        <li class="nav-item">
+                            <a id="@menuItem.Id"
+                               class="nav-link @activeClass @string.Join(" ", menuItem.Classes)"
+                               @if (!menuItem.LocalNav)
+                               {
+                                   <text>target="_blank"</text>
+                               }
+                               href="@(menuItem.Href ?? "#")">@menuItem.Text</a>

```

Comment:
@DAud-IcI Why would we render a `#` for an empty link? Why not leave the attribute empty?

See https://www.w3.org/2001/tag/2011/01/HashInURI-20110115#NakedHash for some input.

Reply:
There are some interesting answers [here](https://stackoverflow.com/q/5637969/177710), too. To be clear, I've never been a fan of `#` in `href`s.

Reply:
For some reason I thought that would not render the attribute at all. Maybe that's what happens when the value is null, or is that in Vuejs? Not sure.

Anyway, It's fixed, I committed it into `issue/SNOW-12` since this PR is already merged. While I have no issues with additional reviews please limit it to the ones that are still open, otherwise it's a pain.

Reply:
But you know what? Have the `#`. We have more important things to do. It's just a fallback, anyways.

Reply:
Sure thing. Will stick to open issues.